wpilibj/src/main/java/edu/wpi/first/wpilibj/controller/PIDController.java

```diff
@@ -422,6 +359,23 @@ protected double getContinuousError(double error) {
     return error;
   }
 
+  /**
+   * Sets the minimum and maximum values expected from the input.
+   *
+   * @param minimumInput The minimum value expected from the input.
+   * @param maximumInput The maximum value expected from the input.
+   */
+  private void setInputRange(double minimumInput, double maximumInput) {
```

Comment:
Does this method need to be public to handle the case where the input range is not continuous?

Reply:
No. It was only ever used for percent tolerance and internally for continuous input, and we're removing percent tolerance.

Reply:
It is used as part of `setSetpoint`.  Teams use it as a way to set super soft limits on mechanisms.

Reply:
If they set their setpoint in the wrong range, that's user error.

Reply:
It's also worth noting that none of the other controller stuff has this functionality because it's extra work no one uses.

Reply:
The more correct way to impose soft limits is to prevent the motor from driving in a certain direction when an encoder threshold has been crossed.

Reply:
Something I have seen is teams using a joystick to change the set point target of an elevator.  They then use the input range argument to clamp the value.

Reply:
Like this:
```
double target = 0.0;

public void update() {
  target += joystick.getRawAxis(0);
  pidController.setSetpoint(target);
}
```

Reply:
That doesn't really belong in PIDController.

Reply:
If they must do the profile manually, they can do this:
```java
double target = 0.0;

public void update() {
  double delta = joystick.getRawAxis(0) * dt;
  target = Math.max(kMinSetpoint, Math.min(target + delta, kMaxSetpoint));
  pidController.setSetpoint(target);
}
```

or in C++, this:
```cpp
double target = 0.0;

void Update() {
  double delta = joystick.GetRawAxis(0) * dt;
  target = std::clamp(target + delta, kMinSetpoint, kMaxSetpoint);
  pidController.setSetpoint(target);
}
```
One should validate input as early as possible for separation of responsibilities.